packages/mui-material/src/Menu/Menu.js

```diff
@@ -259,6 +280,25 @@ Menu.propTypes /* remove-proptypes */ = {
    * `classes` prop applied to the [`Popover`](/material-ui/api/popover/) element.
    */
   PopoverClasses: PropTypes.object,
+  /**
+   * The extra props for the slot components.
+   * You can override the existing props or add new ones.
+   *
+   * @default {}
+   */
+  slotProps: PropTypes.shape({
+    paper: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+    root: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+  }),
+  /**
+   * The components used for each slot inside.
+   *
+   * @default {}
+   */
+  slots: PropTypes.shape({
+    paper: PropTypes.elementType,
+    root: PropTypes.elementType,
+  }),
```

Comment:
The proptypes should be generated from `Menu.d.ts`:

```js
Menu.propTypes /* remove-proptypes */ = {
  // ----------------------------- Warning --------------------------------
  // | These PropTypes are generated from the TypeScript type definitions |
  // |     To update them edit the d.ts file and run "yarn proptypes"     |
  // ----------------------------------------------------------------------
```

So here we should remove these, add `slotProps` and `slots` to `Menu.d.ts` and then run `yarn proptypes`. The result should be the same but that way we have the same in the typescript declaration as well as in the proptypes.

You can follow the Popover type file as reference: https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Popover/Popover.d.ts#L111-L129